### PR TITLE
Ds/vancouver harbor

### PR DESCRIPTION
**Pull request by dsteinmo**

### CODE REVIEW

File: src/MeshManager.cpp [hunk 1]
```diff
@@ -227,11 +227,27 @@ namespace blitzdg {
...
+        blitz::firstIndex ii;
+        BCTable = 0*ii;
+        for (index_type k=0; k < NumElements; ++k) {
+            for (index_type f=0; f < 3; ++f) {
+                index_type k2 = E2E(k*NumFaces + f);
+
+                // self-referencing elements are boundary elements.
+                if (k2 == k)
+                    BCTable(3*k + f) = tagNumber;
+            }
```
Comment (k7miller) [inline src/MeshManager.cpp]: Is the magic number 3 equal to `NumFaces` here? If so then you can replace the nested loops with a single loop which would be more efficient. For the test `k2 == k` you could use `k2 == k / NumFaces`, where `k` now ranges over `0,...,NumFaces*NumElements-1`.

File: src/MeshManager.cpp [hunk 2]
```diff
@@ -451,6 +467,12 @@ namespace blitzdg {
 
     void MeshManager::readElements(const string& E2VFile) {
         EToV = csvread<index_type>(E2VFile, NumElements, NumFaces);
+
+        BCType = index_vec_smart_ptr(new index_vector_type(NumElements*NumFaces));
+        EToE = index_vec_smart_ptr(new index_vector_type(NumElements*NumFaces));
+        EToF = index_vec_smart_ptr(new index_vector_type(NumElements*NumFaces));
+        buildConnectivity();
+        buildBCTable(3);
     }
```
Comment (k7miller) [inline src/MeshManager.cpp:481]: Same as above, does 3 equal `NumFaces`?

File: src/TriangleNodesProvisioner.cpp [hunk 3]
```diff
@@ -507,8 +507,12 @@ namespace blitzdg {
         n_y = n_y/norm;
 
         for(index_type k=0; k < NumElements; ++k) {
-            for (index_type i=0; i < NumFacePoints; i++) {
-                Fscal = norm/(Jac(Fmsk(i),k));
+            index_type count=0;
+            for (index_type f=0; f < NumFaces; ++f) {
+                for (index_type n=0; n < NumFacePoints; n++) {
+                    Fscal(count,k) = norm(count,k)/(Jac(Fmsk(n,f),k));
+                    ++count;
```
Comment (k7miller) [inline src/TriangleNodesProvisioner.cpp:548]: Ideally `Fscal`, `norm`, `Jac`, and `Fmsk` are all column-major here...

**Review by k7miller — COMMENTED** [review]
Looks pretty good bud, just a few comments.

File: src/MeshManager.cpp (hunk 2 above)
Comment (dsteinmo) [reply src/MeshManager.cpp:481]: On line 475, 3 refers to a tag number. I've added the long-awaited enum to clear this up.

File: src/TriangleNodesProvisioner.cpp (hunk 3 above)
Comment (dsteinmo) [reply src/TriangleNodesProvisioner.cpp:548]: kk, got it.

**Comment by dsteinmo** [comment]
:shipit:

**Pull request merged**